### PR TITLE
Add pyvrp.educate.neighbourhood to the docs, add neighbourhood constructor to local search

### LINKED ISSUE #105: Document pyvrp.educate.neighbourhood

### PR DESCRIPTION
**Pull request by N-Wouda**
This PR:

- [x] Closes #105.
- [x] Adds and passes relevant tests.
- [x] Has well-formatted code and documentation.

<details>

**Notes**:

- It is essential that you add a test when making code changes.
  This keeps the code coverage level up, and helps ensure the changes work as intended.
- When fixing a bug, you must add a test that would produce the bug in the master branch, and then show that it is fixed with the new code. 
- New code additions must be well formatted. Changes should pass the pre-commit workflow, which you can set up locally using [pre-commit](https://pre-commit.com/#intro). 
- Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

File: pyvrp/cpp/educate/LocalSearch_bindings.cpp [hunk 1]
```diff
@@ -9,10 +9,14 @@ namespace py = pybind11;
 PYBIND11_MODULE(LocalSearch, m)
 {
     py::class_<LocalSearch>(m, "LocalSearch")
-        .def(py::init<ProblemData &, PenaltyManager &, XorShift128 &>(),
+        .def(py::init<ProblemData &,
+                      PenaltyManager &, 
+                      XorShift128 &,
+                      std::vector<std::vector<int>>>(),
```
Comment (wouterkool) [inline pyvrp/cpp/educate/LocalSearch_bindings.cpp:15]: Not sure, but can't you use the `LocalSearch::Neighbours` type here?

File: pyvrp/educate/tests/test_LocalSearch.py [hunk 2]
```diff
@@ -61,7 +83,10 @@ def test_local_search_set_get_neighbours(
     seed = 42
     rng = XorShift128(seed=seed)
     pen_manager = PenaltyManager(data.vehicle_capacity)
-    ls = LocalSearch(data, pen_manager, rng)
+
+    prev_neighbours = [[client] for client in range(data.num_clients)]
```
Comment (wouterkool) [inline pyvrp/educate/tests/test_LocalSearch.py:87]: This makes me think: shouldn't we validate that client can't be a neighbour of client itself? I think u = v gives undefined behavior in the local search. (I understand that will not be the case here if you prepend the depot afterwards.)

**Review by wouterkool — APPROVED** [review]
LGTM

File: pyvrp/cpp/educate/LocalSearch_bindings.cpp (hunk 1 above)
Comment (N-Wouda) [reply pyvrp/cpp/educate/LocalSearch_bindings.cpp:15]:
```
allocator<int> >, std::allocator<std::vector<int, std::allocator<int> > > >’ is private within this context
   12 |         .def(py::init<ProblemData &,
      |                  ^~~~~~~~~~~~~~~~~~~
   13 |                       PenaltyManager &,
      |                       ~~~~~~~~~~~~~~~~~
   14 |                       XorShift128 &,
      |                       ~~~~~~~~~~~~~~
   15 |                       LocalSearch::Neighbours>(),
      |                       ~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../pyvrp/cpp/educate/LocalSearch_bindings.cpp:1:
../pyvrp/cpp/educate/LocalSearch.h:20:11: note: declared private here
   20 |     using Neighbours = std::vector<std::vector<int>>;
      |           ^~~~~~~~~~
ninja: build stopped: subcommand failed.
```
So I don't think so? We could make it public, but I'm not sure if that helps.

File: pyvrp/educate/tests/test_LocalSearch.py (hunk 2 above)
Comment (N-Wouda) [reply pyvrp/educate/tests/test_LocalSearch.py]: I've added some checks for this (depot and the client itself).

**Pull request merged**